environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java

```diff
@@ -19,59 +19,202 @@
+        }
+        WeldContainer weldContainer = new WeldContainer(id, manager, bootstrap);
+        singleton.set(id, weldContainer);
+        WeldSELogger.LOG.weldContainerInitialized(id);
+        return weldContainer;
+    }
+
+    // This replicates org.jboss.weld.Container.contextId
+    private final String id;
+
+    private final WeldManager manager;
+
+    private final Bootstrap bootstrap;
+
+    private final Instance<Object> instance;
+
+    private final ExperimentalEvent<Object> event;
+
+    /**
+     *
+     * @param id
+     * @param manager
+     * @param bootstrap
+     */
+    private WeldContainer(String id, WeldManager manager, Bootstrap bootstrap) {
+        this.id = id;
+        this.manager = manager;
+        this.bootstrap = bootstrap;
+        this.instance = manager.instance();
+        this.event = ((BeanManagerImpl) manager.unwrap()).event();
+    }
+
+    /**
+     * Provides access to all beans within the application. Retained for backward compatibility.
+     *
+     * @return the instance
      */
     public Instance<Object> instance() {
-        return instanceManager.getInstances();
+        return instance;
     }
 
     /**
      * Provides access to all events within the application. For example:
+     *
      * <code>
-     * weld.event().select(Bar.class).fire(new Bar());
+     * weldContainer.event().select(Bar.class).fire(new Bar());
      * </code>
+     *
+     * @return the event
      */
     public ExperimentalEvent<Object> event() {
-        return instanceManager.getEvents();
+        return event;
+    }
+
+    /**
+     * Weld containers must have a unique identifier assigned when there are multiple Weld instances running at once.
+     *
+     * @return the container id
+     */
+    public String getId() {
+        return id;
     }
 
     /**
      * Provides direct access to the BeanManager.
      */
     public BeanManager getBeanManager() {
-        return beanManager;
+        return manager;
+    }
+
+    /**
+     * Shutdown the container.
+     */
+    public synchronized void shutdown() {
+        if (isRunning()) {
+            try {
+                manager.fireEvent(new Object(), DestroyedLiteral.APPLICATION);
+            } finally {
+                singleton.clear(id);
+                bootstrap.shutdown();
+                WeldSELogger.LOG.weldContainerShutdown(id);
+            }
+        } else {
+            WeldSELogger.LOG.weldContainerAlreadyShutDown(id);
+            if (WeldSELogger.LOG.isTraceEnabled()) {
+                WeldSELogger.LOG.tracev("Spurious call to shutdown from: {0}", (Object[]) Thread.currentThread().getStackTrace());
+            }
+        }
+    }
+
+    /**
+     *
+     * @return <code>true</code> if the container was not shut down yet, <code>false</code> otherwise
+     */
+    public boolean isRunning() {
+        return singleton.isSet(id);
     }
-}
+
+    @Override
+    public void close() throws Exception {
```

Comment:
I don't understand. I've just followed `AutoCloseable.close()` signature. Users will not use/implement this method anyway.